django/db/models/expressions.py

```diff
@@ -651,6 +654,39 @@ def relabeled_clone(self, relabels):
         return self
 
 
+class SliceableF(F):
+    """
+    An object that contains a slice of an F expression.
+
+    Object resolves the column on which the slicing is applied, and then
+    applies the slicing if possible.
+    """
+    def __init__(self, f_obj, slice_obj):
+        if isinstance(slice_obj, int):
+            if slice_obj < 0:
+                raise ValueError('Negative indexing is not supported.')
+            self.start = slice_obj + 1
+            self.length = 1
+        elif isinstance(slice_obj, slice):
+            if (
+                (slice_obj.start is not None and slice_obj.start < 0) or
+                (slice_obj.stop is not None and slice_obj.stop < 0)
+            ):
+                raise ValueError('Negative indexing is not supported.')
+            if slice_obj.step is not None:
+                raise ValueError('Step argument is not supported.')
+            self.start = 1 if slice_obj.start is None else int(slice_obj.start) + 1
+            self.length = None if slice_obj.stop is None else int(slice_obj.stop) - self.start + 1
```

Comment:
Hmm. I guess we probably want to disallow `slice.stop < slice.start` if we're doing these sorts of calculations.

That is unless there are cases where this makes sense for substrings or array indexes. If there are we need to add some testing to handle that.

Reply:
I don't think it makes sense for strings unless you were doing negative indexing which this PR doesn't allow. 

For an array it does not seem to make much sense either as an empty array is returned, see [here](https://dbfiddle.uk/?rdbms=postgres_13&fiddle=74d5dde4ec1ed3743e34551326da323d). (Not sure if I've got my terminology quite right here).

Reply:
I've assumed that we don't need the `int()` conversion. ... If someone's doing [3.4:5.764] is it ok to assume they deserve what they get?